### PR TITLE
Fix close issue but time watcher still running (#17643)

### PR DESCRIPTION
**Pull request by lunny**
back port from #17643

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

File: models/issue_stopwatch.go [hunk 1]
```diff
@@ -74,91 +94,141 @@ func hasUserStopwatch(e Engine, userID int64) (exists bool, sw *Stopwatch, err e
...
+	}
+
+	if _, err := e.Insert(tt); err != nil {
 		return err
 	}
 
-	if exists {
-		// Create tracked time out of the time difference between start date and actual date
-		timediff := time.Now().Unix() - int64(sw.CreatedUnix)
+	if err := issue.loadRepo(x); err != nil {
```
Comment (zeripath) [inline models/issue_stopwatch.go]:
```suggestion
	if err := issue.loadRepo(e); err != nil {
```

**Review by zeripath — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@zeripath done.

**Comment by zeripath** [comment]
Do we need to fix main too?

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
There are many ways in which races resulting in a double stopwatches could still happen here.

Realistically the decision to finish or open a new stopwatch should be done within a transaction with select for updates etc.

However this is a backport and this is just yet another example of easily induceable inconsistency under load.

We have to address this at some point though.

**Pull request merged**